src/main/java/soconnect/model/todo/Todo.java

```diff
@@ -2,40 +2,61 @@
 
 import static soconnect.commons.util.CollectionUtil.requireAllNonNull;
 
+import java.util.Collections;
+import java.util.HashSet;
 import java.util.Objects;
+import java.util.Set;
+
+import soconnect.model.tag.Tag;
 
 /**
- * Represents a Todo in the SoConnect.
- * Guarantees: description is not null and not an empty string.
+ * Represents a {@code Todo} in the SoConnect.
  */
 public class Todo {
-    private final String description;
+    private final Description description;
+    private final Priority priority;
+    private final Set<Tag> tags = new HashSet<>();
 
     /**
-     * Constructs a Todo.
-     *
-     * @param description Must not be null.
+     * Constructs a {@code Todo}. All parameters must be present and not null.
      */
-    public Todo(String description) {
-        requireAllNonNull(description);
+    public Todo(Description description, Priority priority, Set<Tag> tags) {
+        requireAllNonNull(description, priority, tags);
         this.description = description;
+        this.priority = priority;
+        this.tags.addAll(tags);
     }
```

Comment:
Hmmm, not sure if we should support multiple tags since the purpose of this todo is to track the assignments in school, so each todo is related to one module (i.e tag) only.

Also based on your implementation, the tag will be optional, so will it pass the `requireAllNonNull(description, priority, tags)` check?

Reply:
I think it might be useful to have multiple tags, for assignments and tasks that are related to 2 or more modules?
For example: `Watched recorded lecture videos`; tagged under `CS2100` and `ST2334`. Instead of 2 `Todo` with the same description but different tags.

Reply:
All `Set<Tag>` in SoConnect are initialised as empty Sets. So when the user doesn't include any tags, it won't be null but an empty set instead. You can refer to `ParserUtil.parseTags` for confirmation.

Reply:
Ohhh I see, I was thinking the description for the todos should be more specific like `Watch recorded CS2100 lecture videos` and then the user adds a tag to it. Because if there are multiple assignments, but we just create one todo and add many tags to it, it will give the user the impression that he/she does not have that much work to do.

Also about extensibility, if we have a mark/unmark feature in the future, this cannot mark the work when you are combining all the similar todos together. Although extensibility is not a consideration for our project hahaha.